### ISSUE TITLE
MCP does not add javadocs in certain cases

### ISSUE THREAD

**Issue by Pokechu22** [ModCoderPack/MCPBot-Issues#590]
MCP 940 (w/ 1.12.2) does not appear to correctly add javadocs to fields that have a generic type (the same issue probably also applies to methods).  An example of this is <ruby><rb>`EntityLiving.AI_FLAGS`</rb><rp> (</rp><rt>`field_184240_ax`</rt><rp>)</rp></ruby>, which has docs in ForgeGradle but not in MCP.

Likely caused by the regex used, will investigate further as I also investigate MinecraftForge/ForgeGradle#462 (which uses a different regex).

**Comment by Pokechu22** [comment]
Also affects methods (for instance, <ruby><rb>`World.getEntitiesWithinAABB(Class, AxisAlignedBB)`</rb><rp> (</rp><rt>`func_72872_a`</rt><rp>)</rp></ruby>).

Worse yet, also affects nested classes (for instance, `BlockWorkbench$InterfaceCraftingTable.getName`).  That probably is because of the deeper indentation, which explains the FG issue.

**Comment by Pokechu22** [comment]
Fix (using the regex from FG including my fix in MinecraftForge/ForgeGradle#463):

```diff
diff --git a/runtime/commands.py b/runtime/commands.py
index 6fba41e..66e07c2 100755
--- a/runtime/commands.py
+++ b/commands_.py
@@ -1988,8 +1988,8 @@ class Commands(object):
                 fields[row['searge']] = row['desc'].replace('*/', '* /')
 
         regexps = {
-            'field': re.compile(r'^(?P<indent> {4}|\t)(?:[\w$.[\]]+ )*(?P<name>field_[0-9]+_[a-zA-Z_]+) *(?:=|;)'),
-            'method': re.compile(r'^(?P<indent> {4}|\t)(?:[\w$.[\]]+ )*(?P<name>func_[0-9]+_[a-zA-Z_]+)\('),
+            'field': re.compile(r'^(?P<indent>(?: {4})+|\t+)(?!return)(?:\w+\s+)*(?:\w+[\w$.]*(?:<[\w\W]*>)?[\[\]]*)\s+(?P<name>field_[0-9]+_[a-zA-Z_]+) *(?:=|;)'),
+            'method': re.compile(r'^(?P<indent>(?: {4})+|\t+)(?!return)(?:\w+\s+)*(?P<generic><[\w\W]*>\s+)?(?P<return>\w+[\w$.]*(?:<[\w\W]*>)?[\[\]]*)\s+(?P<name>func_[0-9]+_[a-zA-Z_]+)\('),
         }
         wrapper = TextWrapper(width=120)
 ```

I'm currently running some tests to check this.

**Comment by Pokechu22** [comment]
Here's [a diff](https://gist.github.com/Pokechu22/8050c7978a26afef77a917ffc6d42669) of all the comments that this change fixes.
